src/components/Button/Button.tsx

```diff
@@ -61,6 +61,24 @@ export class Button extends React.Component<IButtonProps, IButtonState> implemen
       ? <span className='ms-u-screenReaderOnly' id={ ariaDescriptionId }>{ ariaDescription }</span>
       : null;
 
+    // Check for ariaDescription, description or aria-describedby in the native props to determine source of aria-describedby
+    // otherwise default to null.
+    const getAriaDescribedBy = () => {
```

Comment:
Also I don't really understand why we support both ariaDescriptionId as well as descriptionId.